src/gui/addnewtorrentdialog.cpp

```diff
@@ -94,6 +95,8 @@ AddNewTorrentDialog::AddNewTorrentDialog(QWidget *parent)
     connect(ui->browseButton, SIGNAL(clicked()), SLOT(browseButton_clicked()));
     ui->defaultSavePathCheckBox->setVisible(false); // Default path is selected by default
 
+    ui->doNotDeleteTorrentCheckBox->setVisible(TorrentFileGuard::deleteAddedTorrents());
```

Comment:
It would have been disabled if the option is not enabled. Why take space by a disabled checkbox? The dialog is  already quite complex.